features/bmi_task_features.py

```diff
@@ -4,13 +4,76 @@
 import time
 import numpy as np
 from riglib.experiment import traits, experiment
+from riglib.bmi import clda
 
 ###### CONSTANTS
 sec_per_min = 60
 
 ########################################################################################################
 # Decoder/BMISystem add-ons
 ########################################################################################################
+class RandomUnitDropout(traits.HasTraits):
+    '''
+    Randomly removes units from the decoder. Does not work with CLDA turned on. Units are removed at the
+    end of the delay period on each trial and replaced when the trial ends (either in reward or penalty).
+    The same units will be dropped on repeated trials. The units to drop are specified in the 
+    `unit_drop_groups` attribute by a list of lists of unit indices. The `unit_drop_targets` attribute
+    specifies the target indices on which to drop each group of units.
+    '''
+
+    unit_drop_prob = traits.Float(0, desc="Probability of dropping a group of units from the decoder")
+    unit_drop_groups = traits.Array(value=[[0, 1], [2]], desc="Groups of unit indices to drop from the decoder one at a time")
+    unit_drop_targets = traits.Array(value=[1, 2], desc="Target indices on which to drop groups of units from the decoder")
+
+    def init(self):
+        super().init()
+        self.decoder_units_dropped = np.ones((len(self.decoder.units),), dtype='bool')
+        new_dtype = np.dtype(self.trial_dtype.descr + [('decoder_units_dropped', '?', self.decoder_units_dropped.shape)])
+        self.trial_dtype = new_dtype
+        self.unit_drop_group_idx = 0
+
+        # Save a copy of the decoder
+        self.decoder_orig = self.decoder.copy()
+
+    def _start_wait(self):
+
+        # Decide which units to drop in this trial but don't actually drop them yet
+        if (self.gen_indices[self.target_index] == self.unit_drop_targets[self.unit_drop_group_idx] and 
```

Comment:
I think this should in 'in' instead of '==' in case there are multiple targets to drop a unit on.